### PR TITLE
doc: Describe noalert keyword

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10606 

Issue: 6685

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6685](https://redmine.openinfosecfoundation.org/issues/6685)

Describe changes:
- Document `noalert` keyword.

Update:
- Address review comments

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: doc/userguide/rules/intro.rst [hunk 1]
```diff
@@ -280,6 +280,35 @@ keywords.
...
 Some generic details about keywords follow.
 
+Disabling Alerts
+~~~~~~~~~~~~~~~~
+There is a way to disable alert generation for a rule using the keyword ``noalert``.
+When this keyword is part of a rule, no alert is generated if the other
+portions of the rule match. That is, the other rule actions will *still be
+applied.* Using ``noalert`` can be helpful when a rule is
+collecting or setting state using `flowbits`, `datasets` or other
+state maintenance constructs of the rule language.
```
Comment (jufajardini) [inline doc/userguide/rules/intro.rst:290]: Suggestion: add a reference to the thresholding section, for those interested in controlling alert frequency: https://docs.suricata.io/en/latest/rules/thresholding.html#thresholding-keywords

**Review by jufajardini — CHANGES_REQUESTED** [review]
Mostly looking good, the rule containers are a nice add :)

Two inline comments with possible improvements.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.ftp | 33163 | 34208 | 103.15% | 

Pipeline 19903

File: doc/userguide/rules/intro.rst [hunk 2]
```diff
@@ -280,6 +280,35 @@ keywords.
...
+* The second rule generates an alert if the state is set and additional
+  qualifications are met.
+
+.. container:: example-rule
+
+    :example-rule-action:`alert` :example-rule-header:`http any any -> $HOME_NET any` :example-rule-options:`(msg:"noalert example: set state"; flow:established,to_server; xbits:set,SC.EXAMPLE,track ip_dst, expire 10; noalert; http.method; content:"GET"; sid:1; )`
+
+    :example-rule-action:`alert` :example-rule-header:`http any any -> $HOME_NET any` :example-rule-options:`(msg:"noalert example: state use"; flow:established,to_server; xbits:isset,SC.EXAMPLE,track ip_dst; http.method; content:"POST"; sid: 2; )`
+
+In IPS mode, ``noalert`` is commonly used when Suricata to `drop` network packets
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply doc/userguide/rules/intro.rst:304]: Does "when Suricata should drop [packets]" read better?

File: doc/userguide/rules/intro.rst (hunk 1 above)
Comment (jlucovsky) [reply doc/userguide/rules/intro.rst:290]: Good point.

File: doc/userguide/rules/intro.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/rules/intro.rst:304]: Yes, I'd say so, thanks :)

**Comment by jlucovsky** [comment]
Continued in #10822

**Pull request closed without merging**